QSB/Utility/PlayerToolsManager.cs

```diff
@@ -0,0 +1,36 @@
+﻿using OWML.ModHelper.Events;
+using QSB.Animation;
+using UnityEngine;
+
+namespace QSB.Utility
+{
+    class PlayerToolsManager
+    {
+        private static Transform _cameraBody;
+        private static readonly Vector3 FlashlightOffset = new Vector3(0.7196316f, -0.2697681f, 0.3769455f);
+
+        public static void Init(Transform camera)
+        {
+            _cameraBody = camera;
+
+            CreateFlashlight();
+        }
+
+        private static void CreateFlashlight()
+        {
+            var flashlightRoot = GameObject.Instantiate(GameObject.Find("FlashlightRoot"));
+            flashlightRoot.SetActive(false);
+            var oldComponent = flashlightRoot.GetComponent<Flashlight>();
+            var component = flashlightRoot.AddComponent<QSBFlashlight>();
```

Comment:
what's the point of the QSB version of flashlight?

Reply:
the normal flashlight can't be turned on/off without reflection (and would also control itself which is bad)

Reply:
we want all the on/off logic to be performed on the local side, then send that info to the server

Reply:
wouldn't it be simpler to use some reflection and maybe empty a method or two to disable the automatic stuff?

Reply:
I think it makes sense to have QSBFlashlight, but maybe it should be a much simpler flashlight than the game's.

Maybe we shouldn't just instantiate the entire flashlight object, and instead have only an object with one light source and nothing more. Then you can just enable / disable that object for toggling the flashlight.

The current in-game flashlight contains multiple light sources (maybe just 2, but at least 2) and casts shadows, so having lots of players with lots of flashlights might cause performance issues.

Would be better to save this for a future PR, though.

And @amazingalek patching methods would be difficult because you would also affect the local player's flashlight.